[CodeComplete] Skip conjunction elements unrelated to code completion token 🚥 #59944

ahoppen opened the pull request:
Depends on https://github.com/apple/swift/pull/59944

---

This should improve performance of code completion inside result builders by not type checking conjunction elements that don’t affect code completion.

ahoppen committed:
[CodeCompletion] Disallow void when code completion expr is a solution application target whose result is not unused

This happens if the code completion expression initializes a variable in a multi-statement closure that doesn’t have LeaveClosureBodiesUnchecked set.

ahoppen commented:
@swift-ci Please SourceKit stress test

ahoppen commented:
@swift-ci Please SourceKit stress test

ahoppen commented:
@swift-ci Please smoke test macOS

ahoppen commented:
@swift-ci Please SourceKit stress test

ahoppen committed:
[CodeCompletion] Don’t pass LeaveClosureBodiesUnchecked for solver-based code completion

ahoppen commented:
@swift-ci Please smoke test macOS

ahoppen commented:
@swift-ci Please SourceKit stress test

ahoppen committed:
[CodeCompletion] Use type from solution to determine actor isolation

ahoppen commented:
https://github.com/apple/swift-syntax/pull/723

@swift-ci Please smoke test macOS

ahoppen commented:
https://github.com/apple/swift-syntax/pull/723

@swift-ci Please SourceKit stress test

ahoppen commented:
https://github.com/apple/swift-syntax/pull/723

@swift-ci Please smoke test macOS

ahoppen commented:
https://github.com/apple/swift-syntax/pull/723

@swift-ci Please SourceKit stress test

ahoppen commented:
https://github.com/apple/swift-syntax/pull/723

@swift-ci Please SourceKit stress test

ahoppen commented:
https://github.com/apple/swift-syntax/pull/723

@swift-ci Please smoke test macOS

ahoppen commented:
@swift-ci Please smoke test

ahoppen commented:
@swift-ci Please SourceKit stress test

ahoppen commented:
@swift-ci Please smoke test

ahoppen commented:
@swift-ci Please SourceKit stress test

ahoppen commented:
@swift-ci Please smoke test

ahoppen commented:
@swift-ci Please SourceKit stress test

xedin commented on lib/Sema/CSSyntacticElement.cpp:
```diff
@@ -995,6 +995,10 @@ class SyntacticElementConstraintGenerator
         }
       }
 
+      if (cs.isForCodeCompletion() && !cs.containsIDEInspectionTarget(element)) {
```
Nit: could you please a comment here with brief explanation why this is necessary?

xedin reviewed (approved):
LGTM! I think this is worth a cherry-pick to 5.8

ahoppen committed:
[CodeCompletion] Skip conjuction elements that are unrelated to the code completion token

ahoppen committed:
[CodeCompletion] Skip conjuction elements that are unrelated to the code completion token

ahoppen commented:
@swift-ci Please SourceKit stress test

ahoppen commented:
@swift-ci Please SourceKit stress test

ahoppen commented:
@swift-ci Please SourceKit stress test

ahoppen commented:
@swift-ci Please SourceKit stress test

ahoppen commented:
@swift-ci Please SourceKit stress test

xedin committed:
[CSGen] Detect nested out-of-scope variables in recursive declarations

Follow-up to https://github.com/apple/swift/pull/63505

Since, when the type is not stated, a variable assumes the
type of its initializer that enables out-of-scope variables
to be nested inside of some other concrete type i.e. Optional.

Resolves: https://github.com/apple/swift/issues/63455

ahoppen committed:
[CodeCompletion] Fix issue when completing inside a switch in a closure

ahoppen committed:
[CodeComplete] Fix crash when completing in a string literal in a result builder

ahoppen commented:
@swift-ci Please SourceKit stress test

ahoppen commented:
@swift-ci Please SourceKit stress test

ahoppen commented:
This is now part of https://github.com/apple/swift/pull/63717
